### ISSUE TITLE
Error creating vector layer that conforms to RFC7946 if the geometry crosses both the antimeridian and the prime meridian

### ISSUE THREAD

**Issue by drtimes** [OSGeo/gdal#1068]
## Expected behavior and actual behavior.

An exception is thrown when trying to create a vector layer that conforms to GeoJSON specification (RFC 7946) if the geometry crosses both the antimeridian and the prime meridian (Greenwich).

But according to the standard "Any geometry that crosses the antimeridian SHOULD be represented by cutting it in two such that neither part's representation crosses the antimeridian." https://tools.ietf.org/html/rfc7946#section-3.1.9

## Steps to reproduce the problem.

```
from osgeo import gdal

gdal.UseExceptions()

gdal.VectorTranslate('/vsimem/out.json', """{
  "type": "FeatureCollection",
  "features": [
      { "type": "Feature", "geometry": {"type":"Polygon","coordinates":[[[-218.671875,-16.636192],[-218.671875,67.94165],[16.171875,67.94165],[16.171875,-16.636192],[-218.671875,-16.636192]]]} },
  ]
}""", format = 'GeoJSON', layerCreationOptions = ['RFC7946=YES', 'WRITE_BBOX=YES'])
```

Error:
`Traceback (most recent call last):
  File "/home/serikova/.PyCharm2018.1/config/scratches/scratch_13.py", line 10, in <module>
    }""", format = 'GeoJSON', layerCreationOptions = ['RFC7946=YES', 'WRITE_BBOX=YES'])
  File "/home/serikova/Libraries_install/gdal-2.3.1/lib/python2.7/dist-packages/osgeo/gdal.py", line 725, in VectorTranslate
    return wrapper_GDALVectorTranslateDestName(destNameOrDestDS, srcDS, opts, callback, callback_data)
  File "/home/serikova/Libraries_install/gdal-2.3.1/lib/python2.7/dist-packages/osgeo/gdal.py", line 3218, in wrapper_GDALVectorTranslateDestName
    return _gdal.wrapper_GDALVectorTranslateDestName(*args)
RuntimeError: Terminating translation prematurely after failed
translation of layer OGRGeoJSON (use -skipfailures to skip errors)`

## Operating system

Ubuntu 16.04 64 bit

## GDAL version and provenance
GDAL 2.3.1, released 2018/06/22, build from source code with Python bindings

**Comment by jratike80** [comment]
The error that I get from OSGeo4W shell (GDAL 2.2.4, released 2018/03/19)

```
ERROR 1: Geometry extent outside of [-180.0,180.0]x[-90.0,90.0] bounds
ERROR 1: Unable to write feature 0 from layer OGRGeoJSON.
ERROR 1: Terminating translation prematurely after failed
translation of layer OGRGeoJSON (use -skipfailures to skip errors)
```

**Comment by drtimes** [comment]
Hello, @rouault 

I've found another case when creation of vector layer fails in my build. Could you please check if this issue was also fixed in [93bd242](https://github.com/OSGeo/gdal/commit/93bd2427dbca537cfa6aa37319ec9e1e0b193934)?
Though I am able to create the geometry using `ogr.CreateGeometryFromJson()`,  it fails with `gdal.VectorTranslate()`.

```
from osgeo import gdal

gdal.UseExceptions()

gdal.VectorTranslate('/vsimem/out.json', """{
  "type": "FeatureCollection",
  "features": [
      { "type": "Feature", "geometry": {"type":"Polygon","coordinates":[[[402.989502,30.211608],[402.989502,30.237713],[403.022461,30.237713],[403.022461,30.211608],[402.989502,30.211608]]]} },
  ]
}""", format = 'GeoJSON', layerCreationOptions = ['RFC7946=YES', 'WRITE_BBOX=YES'])
```

I get such longitudes from Leaflet. I found out that OGR with GEOS fails to compute intersection of the geometries if longitude exceeds limits, so I'm trying to use RFC7946 normalization as a workaround
